components/pkg-export-tar/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+pub const VERSION: &'static str = include_str!(concat!(env!("OUT_DIR"), "/VERSION"));
+
+pub fn export_for_cli_matches(ui: &mut UI, matches: &clap::ArgMatches) -> Result<()> {
+    let default_channel = channel::default();
+    let default_url = hurl::default_bldr_url();
+    let spec = BuildSpec::new_from_cli_matches(&matches, &default_channel, &default_url);
+    export(ui, spec)?;
+
+    Ok(())
+}
+
+pub fn export(ui: &mut UI, build_spec: BuildSpec) -> Result<()> {
+
+    let hart_to_package = build_spec.idents_or_archives.join(", ");
+    let builder_url = build_spec.url;
+
+    ui.begin(format!("Building a tarball with: {}", hart_to_package))?;
+
+    let temp_dir_path = Temp::new_dir().unwrap().to_path_buf();
+
+    initiate_tar_command(&temp_dir_path, &hart_to_package, &builder_url);
+
+    Ok(())
+}
+
+fn initiate_tar_command(temp_dir_path: &PathBuf, hart_to_package: &str, builder_url: &str) {
+    let status = Command::new("hab")
+        .arg("studio")
+        .arg("-r")
+        .arg(&temp_dir_path)
+        .arg("new")
+        .status()
+        .expect("failed to create studio");
+
+    if status.success() {
+        println!("Able to create studio to export package as tarball, proceeding...");
+        install_command(&temp_dir_path, &hart_to_package, &builder_url);
+    } else {
+        println!("Unable to create a studio to export the package as a tarball.")
+    }
+}
+
+fn install_command(temp_dir_path: &PathBuf, hart_to_package: &str, builder_url: &str) {
+    let status = Command::new("hab")
+        .arg("studio")
+        .arg("-q")
+        .arg("-r")
+        .arg(&temp_dir_path)
+        .arg("run")
+        .arg("hab")
+        .arg("install")
+        .arg("-u")
+        .arg(builder_url)
+        .arg(&hart_to_package)
+        .status()
+        .expect("failed to install package in studio");
+
+    if status.success() {
+        println!("Hart package is installable in a studio, exporting tarball...");
+        tar_command(&temp_dir_path, &hart_to_package);
+    } else {
+        println!("Hart package is NOT installable in a studio and could not be exported.");
+        println!("Please see the above error for details.");
+    }
+}
+
+fn tar_command(temp_dir_path: &PathBuf, hart_to_package: &str) {
+    let status = Command::new("tar")
+        .arg("cpzf")
+        .arg(tar_name(&hart_to_package))
+        .arg("-C")
+        .arg(&temp_dir_path)
+        .arg("./hab/pkgs")
+        .arg("./hab/bin")
+        .status()
+        .expect("failed to create tarball");
+
+    if status.success() {
+        println!("Tarball export complete!")
+    } else {
+        println!("Unable to export package to tarball.")
+    }
+}
+
+fn tar_name(hart_to_package: &str) -> String {
+    let path = Path::new(hart_to_package);
+    if path.is_file() {
+        let ident = PackageArchive::new(path).ident().unwrap();
+        format_tar_name(ident)
+    } else {
+        let pkg_path_command = Command::new("hab")
```

Comment:
In this branch, we're presumably dealing with a package identifier. We've actually got code to handle the resolution of a full identifier from a partial one in our core install logic. It will pull down the package in question from Builder if it doesn't already exist in the cache, which the current studio-based implementation in this PR does not do, but it does line up with the behavior in the rest of the project.

Also, just from a separation-of-concerns viewpoint, this function is doing several things: figuring out whether we're dealing with a hart file or an identifier, figuring out what full identifier to use, and then formatting that identifier into a tar file name. 

There's also a hidden, assumed behavior based on how it's called, where `hart_to_package` could actually be a string of multiple identifiers to resolve, but I don't think this code actually handles that behavior. 

And actually, now that I type that, any "multiple" package export would only be named for a single package (presumably the first one), which is potentially confusing... do we really want to take more than one hart file or identifier as an argument to this exporter? (This thought actually applies to our other exporters as well, because I think they have the same assumptions.) @fnichol do you have any thoughts on this point?

Reply:
I had not thought to try it with more than one hart file or identifier.  Does the current tar exporter allow for this?  @fnichol would love your thoughts as well :)